### PR TITLE
[FLINK-26280][table-planner] Add 'table.exec.legacy-transformation-uids' to support old transformation uid generation behaviour

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Per discussion here https://issues.apache.org/jira/browse/FLINK-25932

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
@twalthr seems like when the flag is deprecated, the docs doesn't show the flag (at least, i tried to generate, but nothing changed). Perhaps for this flag is better to not show in the docs? WDYT?

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java [hunk 1]
```diff
@@ -131,15 +133,17 @@ public DynamicTableSourceSpec getTableSourceSpec() {
         } else if (provider instanceof DataStreamScanProvider) {
             Transformation<RowData> transformation =
                     ((DataStreamScanProvider) provider)
-                            .produceDataStream(createProviderContext(), env)
+                            .produceDataStream(
+                                    createProviderContext(planner.getConfiguration()), env)
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java:137]: Use the new `ExecNodeConfig` here. `planner.getConfiguration()` should not be accessed directly anymore.

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks @slinkydeveloper. Can we add at least one test for this? I think `org.apache.flink.table.planner.plan.nodes.exec.TransformationsTest` would be a good location to quickly check a pipeline from source -> one or two operators -> sink.

Not having it in the docs is fine with me.

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java [hunk 2]
```diff
@@ -131,15 +133,17 @@ public DynamicTableSourceSpec getTableSourceSpec() {
...
                             .getTransformation();
             meta.fill(transformation);
             transformation.setOutputType(outputTypeInfo);
             return transformation;
         } else if (provider instanceof TransformationScanProvider) {
             final Transformation<RowData> transformation =
                     ((TransformationScanProvider) provider)
-                            .createTransformation(createProviderContext());
+                            .createTransformation(
+                                    createProviderContext(planner.getConfiguration()));
```
Comment (matriv) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecTableSourceScan.java:146]: same here.

**Review by matriv — APPROVED** [review]
LGTM

**Pull request closed without merging**